tests/functionality_tests/test_activation_shapes.py

```diff
@@ -7,16 +7,43 @@ def main():
 
     grid=Grid(42.444250, 42.444599, -76.483682, -76.483276)
 
+
+    try: 
+        rows = int(input("number of rows:"))
+    except:
+        rows = grid.get_num_rows()
+
+
+    try: 
+        cols = int(input("number of cols:"))
+    except:
+        cols = grid.get_num_cols()
```

Comment:
Does this mean we should remove the input option?

Reply:
yes!

Explanation/reasoning copied from slack for documentation purposes

If we want to keep this as a functionality test we should delete it.

Just for some more context, functionality tests should be scripts testing a specific scenario. In this case the original motivation was to turn Aditya’s interactive script for the activation shapes into a script testing a specific case scenario — in this case the space determine by the lat/long of the eng quad (which converts to 39 columns, 34 rows using our grid code). In an idea case we would just have to run the script and compare it with the expected output/image to make sure it’s working. This idea can be confusing but a good example of this is Allen’s obstacle avoidance script obstacle_tracking_test.py, which mocks a scenario written in csv files, runs the code under that case, and checks if the output is as expected. By having a static cases, we can quickly check even without much understanding + potentially expand to using render tests to automatically check image-based functionality tests (like we do rn on github for the compilation tests).

In the case where we want to determine the bounds rows/cols of the grid, that would be more of a visualizer (if we preset inputs then grid pops up) or a interactive gui/visualizer (if we pop up w grid and allow dynamic changes like changing shape).

Though it is good for debugging and playing around w the code, I would say the try except to determine the number of rows/cols isn’t the clearest interaction for a visualizer since there are a lot of ambiguity (e.g. would need to look at code to know pressing “enter” leads to a default of the entire grid bounds being used, need to tell users to enter valid numbers, etc). Additionally,

```
plt.xlim(-1, grid.get_num_rows()+1)
plt.ylim(-1, grid.get_num_cols()+1)
```

shows the size 39, 34 regardless (which is why the circle case Shreya brought up becomes off screen.
I think it would be good to create a visualizer later on since it’ll require more changes than keeping this as a functionality script (easy future onboarding task o7). Now that I’m thinking about this again, to be more of a functionality test, it shouldn’t have any user input even for the shapes and instead be in separate files or functions or something — but lets make that a later issues xD

I think the best approach would be to
- Keep this as a functionality test, testing the adjustable grid code on the eng quad
- Separate out each shape into individual scripts
- Create a interactive visualization to allow users to play around and determine what they see